src/components/AddAssignment.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+
+// properties addAssignment is required, function called when Add clicked.
+class AddAssignment extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            open: false,
+        
+            //assignmentId: null,
+            courseId: '',
+            assignmentName: '',
+            dueDate: '',
+            //courseTitle: null
+            //needs_grading: 1,
+        
+        };
+        
+        this.handleChange = this.handleChange.bind(this);
+    };
+    
+    handleClickOpen = () => {
+        this.setState( {open:true} );
+    };
+    
+    handleClose = () => {
+        this.setState( {open:false} );
+    };
+    
+    handleChange = (event) => {
+        this.setState({courseId: event.target.value});
+        this.setState({assignmentName: event.target.value});
+        this.setState({dueDate: event.target.value});
```

Comment:
something here is broken see comment below

Reply:
You may want to have separate methods for each state prop. (i.e., `handleChangeCourseId()`, `handleChangeAssignmentName()`, `handleChangeDueDate()`).

OR, leverage the `name=` attribute of the `<input>` element like I've done in [my code](https://github.com/josevh/CST438-register-frontend-jherreragallegos/blob/master/src/components/AddStudent.js#L33-L42) so you know which field to update.